src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,664 @@
+import ListItem from '../List/ListItem'
+import Icon from '../Icon/Icon'
+import { commonUIComponentPropTypes } from '../../lib/commonPropTypes'
+import Ref from '../Ref/Ref'
+import { UIComponentProps } from 'src/lib/commonPropInterfaces'
+
+// To be replaced when Downshift will add highlightedItem in their interface.
+export interface DownshiftA11yStatusMessageOptions<Item> extends A11yStatusMessageOptions<Item> {
+  highlightedItem: Item
+}
+
+export interface DropdownProps extends UIComponentProps<any, any> {
+  /** The default value for the search query. */
+  defaultSearchQuery?: string
+
+  /** The default value for the dropdown. */
+  defaultValue?: DropdownListItem | DropdownListItem[]
+
+  /** A dropdown can take the full width of its container. */
+  fluid?: boolean
+
+  /**
+   * A function that creates custom accessability message for dropdown status.
+   * @param {Object} messageGenerationProps - Object with properties to generate message from. See getA11yStatusMessage from Downshift repo.
+   */
+  getA11yStatusMessage?: (options: DownshiftA11yStatusMessageOptions<DropdownListItem>) => string
+
+  /**
+   * A function that creates custom accessability message for dropdown item selection.
+   * @param {DropdownListItem} item - Dropdown selected element.
+   */
+  getA11ySelectedMessage?: (item: DropdownListItem) => string
+
+  /** A function that creates custom accessability message for dropdown item removal.
+   * @param {DropdownListItem} item - Dropdown removed element.
+   */
+  getA11yRemovedMessage?: (item: DropdownListItem) => string
+
+  /** A function that creates custom aria label accessability message for the remove item button.
+   * @param {DropdownListItem} item - The active item to be removed.
+   */
+  getA11yRemoveItemMessage?: (item: DropdownListItem) => string
+
+  /** Array of props for generating dropdown items and selected item labels if multiple selection. */
+  items?: DropdownListItem[]
+
+  /**
+   * Function to be passed to create selected searchQuery from selected item. It will be displayed on selection in the
+   * edit text, for search, or on the button, for non-search. Multiple search will always clear searchQuery on selection.
+   */
+  itemToString?: (item: DropdownListItem) => string
+
+  /** A dropdown can perform a multiple selection. */
+  multiple?: boolean
+
+  /** A string to be displayed when dropdown does not have available items to show. */
+  noResultsMessage?: string
+
+  /**
+   * Called on deletion by backspace when dropdown is a multiple search.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All props and proposed value.
+   */
+  onBackspaceDelete?: ComponentEventHandler<DropdownProps>
+
+  /**
+   * Callback for change in dropdown active value(s).
+   * @param {DropdownListItem|DropdownListItem[]} value - Dropdown active value(s).
+   */
+  onDropdownChange?: (value: DropdownListItem | DropdownListItem[]) => any
```

Comment:
this prop should be split on two - one for each autocontrolled prop:
- `onChange` - for value change (no need to repeat `dropdown` word)
- `onSearchQueryChange` - for `searchQuery` changes

Note that the naming convention is:
- 'onChange' - for `value` prop changes
- 'onXChange' - for `X` prop changes

Reply:
I cannot, as I will need to provide an Event in the callback if I will name it 'onChange', and Downshift is not providing me one. :( :(